### PR TITLE
Extended FDB test case

### PR DESCRIPTION
**Pull request by yvolynets-mlnx**
Signed-off-by: Yuriy Volynets <yuriyv@mellanox.com>

### Description of PR

Summary: Extended FDB test case with ARP packets
Fixes # (issue)

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [X] Test case(new/improvement)

#### How did you do it?
Added verification that FDB table is filling in after DUT receive ARP request or ARP reply packets.

#### How did you verify/test it?
Tested on the local setup.

### CODE REVIEW

File: tests/fdb/test_fdb.py [hunk 1]
```diff
@@ -55,15 +100,17 @@ def send_recv_eth(ptfadapter, source_port, source_mac, dest_port, dest_mac):
...
 
-def setup_fdb(ptfadapter, vlan_table, router_mac):
+def setup_fdb(ptfadapter, vlan_table, router_mac, pkt_type):
     """
     :param ptfadapter: PTF adapter object
     :param vlan_table: VLAN table map: VLAN subnet -> list of VLAN members
     :return: FDB table map : VLAN member -> MAC addresses set
     """
-
+    pkt_types = ["ethernet", "arp_request", "arp_reply"]
```
Comment (stepanblyschak) [inline tests/fdb/test_fdb.py]: packet types are coded twice in the code

File: tests/fdb/test_fdb.py [hunk 2]
```diff
@@ -79,8 +126,12 @@ def setup_fdb(ptfadapter, vlan_table, router_mac):
                           for i in range(DUMMY_MAC_COUNT)]
 
             for dummy_mac in dummy_macs:
-                send_eth(ptfadapter, member, dummy_mac, router_mac)
-
+                if pkt_type == "ethernet":
+                    send_eth(ptfadapter, member, dummy_mac, router_mac)
+                elif pkt_type == "arp_request":
+                    send_arp_request(ptfadapter, member, dummy_mac, router_mac)
+                else:
```
Comment (stepanblyschak) [inline tests/fdb/test_fdb.py]: else pkt_type == 'arp_reply'
else: pytest.error('Unknonw')'

File: tests/fdb/test_fdb.py [hunk 3]
```diff
@@ -112,7 +166,7 @@ def test_fdb(ansible_adhoc, testbed, ptfadapter):
         pytest.skip('unsupported testbed type')
 
     duthost = AnsibleHost(ansible_adhoc, testbed['dut'])
-    ptfhost = AnsibleHost(ansible_adhoc, testbed['ptf'])
+    ptfhost = AnsibleHost(ansible_adhoc, testbed['ptf_ip'])
```
Comment (stepanblyschak) [inline tests/fdb/test_fdb.py]: What is this change about?

File: tests/fdb/test_fdb.py [hunk 4]
```diff
@@ -133,12 +187,12 @@ def test_fdb(ansible_adhoc, testbed, ptfadapter):
...
-    for vlan in vlan_table:
-        for src, dst in itertools.combinations(vlan_table[vlan], 2):
-            for src_mac, dst_mac in itertools.product(fdb[src], fdb[dst]):
-                send_recv_eth(ptfadapter, src, src_mac, dst, dst_mac)
+    fdb = setup_fdb(ptfadapter, vlan_table, router_mac, pkt_type)
+    if pkt_type == "ethernet":
+        for vlan in vlan_table:
+            for src, dst in itertools.combinations(vlan_table[vlan], 2):
+                for src_mac, dst_mac in itertools.product(fdb[src], fdb[dst]):
+                    send_recv_eth(ptfadapter, src, src_mac, dst, dst_mac)
```
Comment (stepanblyschak) [inline tests/fdb/test_fdb.py]: Do you observe instability here when you run arp testcase seperately from ethernet?

File: tests/fdb/test_fdb.py (hunk 3 above)
Comment (stepanblyschak) [reply tests/fdb/test_fdb.py]: Please, consider using testbed_devices

File: tests/fdb/test_fdb.py [hunk 5]
```diff
@@ -95,14 +146,17 @@ def fdb_cleanup(ansible_adhoc, testbed):
     duthost = AnsibleHost(ansible_adhoc, testbed['dut'])
     try:
         duthost.command('sonic-clear fdb all')
+        time.sleep(5)
```
Comment (stepanblyschak) [inline tests/fdb/test_fdb.py]: Add a comment descirbing the issue you are W/A with sleep

**Review by stepanblyschak — CHANGES_REQUESTED** [review]
As commented and I don't see a call to pytest from ansible

File: tests/fdb/test_fdb.py (hunk 1 above)
Comment (yvolynets-mlnx) [reply tests/fdb/test_fdb.py]: Fixed.

File: tests/fdb/test_fdb.py (hunk 2 above)
Comment (yvolynets-mlnx) [reply tests/fdb/test_fdb.py]: Fixed.

File: tests/fdb/test_fdb.py (hunk 3 above)
Comment (yvolynets-mlnx) [reply tests/fdb/test_fdb.py]: Fixed.

File: tests/fdb/test_fdb.py (hunk 4 above)
Comment (yvolynets-mlnx) [reply tests/fdb/test_fdb.py]: Yes.

File: tests/fdb/test_fdb.py (hunk 5 above)
Comment (yvolynets-mlnx) [reply tests/fdb/test_fdb.py]: For ARP packets traffic is not verifying, so delays is redundant.

Comment (yvolynets-mlnx) [reply tests/fdb/test_fdb.py]: Fixed.

**Review by stepanblyschak — APPROVED** [review]

**Pull request closed without merging**